readers/messages.go

```diff
@@ -0,0 +1,17 @@
+package readers
+
+import (
+	"errors"
+
+	"github.com/mainflux/mainflux"
+)
+
+// ErrNotFound indicates that requested entity doesn't exist.
+var ErrNotFound = errors.New("entity not found")
+
+// MessageRepository specifies message reader API.
+type MessageRepository interface {
```

Comment:
We can rename `MessageReposistory` (reader as well as writer) to `Reader` and `Writer`.